contracts/src/v0.8/ccip/pools/GHO/UpgradeableLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,250 @@
+    return interfaceId == type(ILiquidityContainer).interfaceId || super.supportsInterface(interfaceId);
+  }
+
+  /// @notice Gets LiquidityManager, can be address(0) if none is configured.
+  /// @return The current liquidity manager.
+  function getRebalancer() external view returns (address) {
+    return s_rebalancer;
+  }
+
+  /// @notice Sets the LiquidityManager address.
+  /// @dev Only callable by the owner.
+  function setRebalancer(address rebalancer) external onlyOwner {
+    s_rebalancer = rebalancer;
+  }
+
+  /// @notice Sets the bridge limit, the maximum amount of tokens that can be bridged out
+  /// @dev Only callable by the owner or the bridge limit admin.
+  /// @dev Bridge limit changes should be carefully managed, specially when reducing below the current bridged amount
+  /// @param newBridgeLimit The new bridge limit
+  function setBridgeLimit(uint256 newBridgeLimit) external {
+    if (msg.sender != s_bridgeLimitAdmin && msg.sender != owner()) revert Unauthorized(msg.sender);
+    uint256 oldBridgeLimit = s_bridgeLimit;
+    s_bridgeLimit = newBridgeLimit;
+    emit BridgeLimitUpdated(oldBridgeLimit, newBridgeLimit);
+  }
+
+  /// @notice Sets the bridge limit admin address.
+  /// @dev Only callable by the owner.
+  /// @param bridgeLimitAdmin The new bridge limit admin address.
+  function setBridgeLimitAdmin(address bridgeLimitAdmin) external onlyOwner {
+    address oldAdmin = s_bridgeLimitAdmin;
+    s_bridgeLimitAdmin = bridgeLimitAdmin;
+    emit BridgeLimitAdminUpdated(oldAdmin, bridgeLimitAdmin);
+  }
+
+  /// @notice Gets the bridge limit
+  /// @return The maximum amount of tokens that can be transferred out to other chains
+  function getBridgeLimit() external view virtual returns (uint256) {
+    return s_bridgeLimit;
+  }
+
+  /// @notice Gets the current bridged amount to other chains
+  /// @return The amount of tokens transferred out to other chains
+  function getCurrentBridgedAmount() external view virtual returns (uint256) {
+    return s_currentBridged;
+  }
+
+  /// @notice Gets the bridge limiter admin address.
+  function getBridgeLimitAdmin() external view returns (address) {
+    return s_bridgeLimitAdmin;
+  }
+
+  /// @notice Checks if the pool can accept liquidity.
+  /// @return true if the pool can accept liquidity, false otherwise.
+  function canAcceptLiquidity() external view returns (bool) {
+    return i_acceptLiquidity;
+  }
+
+  /// @notice Adds liquidity to the pool. The tokens should be approved first.
+  /// @param amount The amount of liquidity to provide.
+  /// @dev amount being added + currentBridged needs to be within the bridge limit, otherwise increase bridge limit first
+  function provideLiquidity(uint256 amount) external {
+    if (!i_acceptLiquidity) revert LiquidityNotAccepted();
+    if (s_rebalancer != msg.sender) revert Unauthorized(msg.sender);
+
+    if ((s_currentBridged += amount) > s_bridgeLimit) revert BridgeLimitExceeded(s_bridgeLimit);
```

Comment:
I d suggest to keep standard functionality of CCIP Token Pools and move the `s_currentBridged` update to a separate permissioned function. Basically, a `setCurrentBridgedAmount` setter

Reply:
hesitant to not uphold bridgedAmount == ghoBalance invariant but updated for flexibility [92c9582](https://github.com/aave/ccip/pull/20/commits/92c9582b098a0775c272ffaac9a383eb9d01faca)